win32ss/user/user32/windows/window.c

```diff
@@ -1512,6 +1512,37 @@ IsIconic(HWND hWnd)
 }
 
 
+/**
+ * @brief
+ * IsServerSideWindow
+ *
+ * Checks whether a window has a window procedure that resides in kernel mode.
+ *
+ * @param[in] hWnd
+ * A handle to the window to be checked.
+ *
+ * @return
+ * TRUE if the window has a window procedure that resides in kernel mode,
+ * FALSE otherwise.
+ *
+ * @remarks
+ * Undocumented, see http://undoc.airesoft.co.uk/user32.dll/IsServerSideWindow.php
+ * (unofficial documentation).
+ */
+BOOL
+WINAPI
+IsServerSideWindow(
+    _In_ HWND hWnd)
+{
+    PWND Wnd = ValidateHwnd(hWnd);
```

Comment:
Note that this function (and thus, your IsServerSideWindow implementation as it is currently written) will set the last-error in case the window handle is invalid. You should probably add a test for this.
If the last-error remains untouched, then please use `ValidateHwndNoErr` instead.